src/features/game/types/milestones.ts

```diff
@@ -0,0 +1,146 @@
+import { getFishByType } from "features/island/hud/components/codex/lib/utils";
+import { KNOWN_IDS } from ".";
+import { BumpkinItem } from "./bumpkin";
+import { getKeys } from "./craftables";
+import { FishType, FishName, FISH } from "./fishing";
+import { InventoryItemName, GameState } from "./game";
+
+export type MilestoneName =
+  | "Novice Angler"
+  | "Advanced Angler"
+  | "Expert Angler"
+  | "Fish Encyclopedia"
+  | "Master Angler";
+
+type MilestoneReward = InventoryItemName | BumpkinItem;
+
+export const isInventoryItemReward = (
+  reward: MilestoneReward
+): reward is InventoryItemName => {
+  return reward in KNOWN_IDS;
+};
+
+export type Milestone = {
+  task: string;
+  percentageComplete: (farmActivity: GameState["farmActivity"]) => number;
+  reward: Partial<Record<MilestoneReward, number>>;
+};
+
+const FISH_BY_TYPE: Record<FishType, FishName[]> = getFishByType();
+
+export const FISH_MILESTONES: Record<MilestoneName, Milestone> = {
```

Comment:
Curious: Are any of the milestone rewards going to confer boosts?  Even small ones like -10% cooking duration for fish recipes would be compelling.